### PR TITLE
Shorten localNetwork.addNode

### PR DESCRIPTION
**Pull request by holisticode**
Closes https://ava-labs.atlassian.net/browse/FRAM-246

This PR also removes the k8s test run as k8s has been demoted in priority for the time being and is not currently actively developed.

### CODE REVIEW

File: local/network.go [hunk 1]
```diff
@@ -737,3 +592,219 @@ func createFileAndWrite(path string, contents []byte) error {
...
+	for flagName, flagVal := range nodeConfig.Flags {
+		if _, ok := warnFlags[flagName]; ok {
+			ln.log.Warn("The flag %s has been provided. This can create conflicts with the runner. The suggestion is to remove this flag", flagName)
+		}
+		flags = append(flags, fmt.Sprintf("--%s=%v", flagName, flagVal))
+	}
+
+	// Parse this node's ID
+	nodeID, err := utils.ToNodeID([]byte(nodeConfig.StakingKey), []byte(nodeConfig.StakingCert))
+	if err != nil {
```
Comment (felipemadero) [inline local/network.go:745]: I should have leave this outside of this function. Probably a new function that generates the ports outside of this, can also return a partial localNode (ports, nodeId). initLocalNode?

File: local/network.go [hunk 2]
```diff
@@ -737,3 +592,219 @@ func createFileAndWrite(path string, contents []byte) error {
...
+}
+
+// buildFlags returns the:
+// 1) Flags
+// 2) API port
+// 3) P2P port
+// 4) Node ID
+// of the node being added with config [nodeConfig], config file [configFile],
+// and directory at [nodeDir].
+func (ln *localNetwork) buildFlags(configFile map[string]interface{}, nodeDir string, nodeConfig *node.Config) (nodeInfo, error) {
```
Comment (felipemadero) [inline local/network.go:686]: should have preferred this function to receive ports (or to receive partial localNode) and return flags, avoiding nodeInfo struct (other function to generate the node ports). name: buildNodeFlags

File: local/network.go [hunk 3]
```diff
@@ -331,49 +336,18 @@ func (ln *localNetwork) AddNode(nodeConfig node.Config) (node.Node, error) {
...
-		ln.log.Warn("no network root directory defined; will create this node's runtime directory in working directory")
-	}
-	// [nodeRootDir] is where this node's config file, C-Chain config file,
-	// staking key, staking certificate and genesis file will be written.
-	// (Other file locations are given in the node's config file.)
-	// TODO should we do this for other directories? Profiles?
-	nodeRootDir := filepath.Join(ln.rootDir, nodeConfig.Name)
-	if err := os.Mkdir(nodeRootDir, 0o755); err != nil {
-		return nil, fmt.Errorf("error creating temp dir: %w", err)
+	nodeDir, err := makeNodeDir(ln.log, ln.rootDir, nodeConfig.Name)
```
[the remark this answers is not in the document]
Comment (danlaine) [reply local/network.go:348]: I prefer using functions over methods if possible

**Review by felipemadero — COMMENTED** [review]
LGTM. Probably would change some few function names to reflect they are related to node and not network stuff. writeNodefiles. buildNodeFlags. But that's a nit anyway.

**Review by felipemadero — APPROVED** [review]

**Pull request merged**